### ISSUE TITLE
Various question about this plugin.. need advice & more explanation

### ISSUE THREAD

**Issue by henkkelder** [j3k0/cordova-plugin-purchase#406]
I've been busy migrating from an older plugin to this one in my cordova/phonegap app. Although I have been searching and reading a lot some thing remain unclear to me.

- First, I see that version 4.1.0 should be available, but issueing 'cordova plugin add ...' keep giving me version 4.0.0. How can I retrieve 4.1.0?

- Then, on Android, after registring all my productID's and calling store.refresh once, the 'whens' for bought non-consumable fire to the state of approved. At that point I check if the payload for the iap has already been downloaded, if not, I launch my download logic. If the payload is already downloaded I call 'finish' on the product and all seems fine. Is this how it should be? (I struggled with this, but found a post from Jean-Christophe  somewhere confirming this).

But on IOS it seems to work different. After registering the products and calling store.refresh my already purchased products reach the 'valid' state but there it stops. Only after a 2nd call to store.refresh the actual state seems to be updated. However this 2nd call also prompts the user to login to the appstore. Is this how it should work? **confusing**  

So I implemented my own mechanism to keep track of what the user already has bought and that seems to work okay but has one disadvantage: It makes it more differcult to detect when a purchase has been returned.

Also the 2nd call to store.refresh gives me fantom products where the id is always 'application data'. As suggested in another post I solved that by not creating general 'product' handlers but only by specific names. 

But all in all it doesn't feel right. Although I do no longer register general when-handlers, under the hood these fantom iaps are still there. 

- I already mentioned it, what is the best approach for handling purchase cancellations? How to I catch these?

**Comment by calendee** [comment]
I'm no expert at this plugin.  I've only implemented it once over a year ago.  I'm working on it again and wondering the same thing about cancelling.

Right now, I'm having success with detecting cancelling during the bug process like this:

```
function buy(product) {

  store.when("product").approved(function(product) {

    product.verify()
      .success(function(product, purchaseData) {
        console.log('Successfully verified!');
        console.log(product);
        console.log(purchaseData);
      })
      .error(function(error) {
        console.log('Verify failed!');
        console.log(error);
      })

  });

  store.when("product").verified(function(product) {
    product.finish();
    // Do stuff here now that purchase was verified
  });

  store.when("product").cancelled(function() {
    // Do stuff here because purchase was cancelled
  });

  store.when("product").error(function() {
    // Do stuff here because purchase failed
  });

  store.when("product").unverified(function(p) {
    // Do stuff here.
  });

  // Iniitiate the purchase
  store.order(product)
});
```

**Comment by henkkelder** [comment]
Thanks Justin. I will implement  "cancelled" handler. 

Any ideas about the differences I believe to see between Android and IOS on the matter of getting the state of already purchased items?

**Comment by calendee** [comment]
@henkkelder No, I haven't even begun Android integration.  I've been dreading it too much.
